### PR TITLE
MGMT-7121: Add vsphere platform support to install-config.yaml

### PR DESCRIPTION
**Pull request by eliorerz**
# Assisted Pull Request

## Description
Support vsphere platform on assisted service https://issues.redhat.com/browse/MGMT-7067

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

### CODE REVIEW

**Comment by gamli75** [comment]
@eliorerz please see @avishayt remark on https://issues.redhat.com/browse/MGMT-7104 - I think we should sync the parameters

**Comment by eliorerz** [comment]
@gamli75 Ithink we can merge this PR. It's just add the ability to assisted-service add platform of type vshpere to the install config

File: internal/installcfg/installcfg.go
Comment (gamli75) [inline internal/installcfg/installcfg.go:44]: Are all those fields are mandatory? maybe we can add remarks for the mandatory fields

Comment (tsorya) [reply internal/installcfg/installcfg.go:44]: @eliorerz can you validate it?

Comment (eliorerz) [reply internal/installcfg/installcfg.go:44]: Just the `Folder` is not mandatory

**Comment by filanov** [comment]
https://docs.openshift.com/container-platform/4.7/installing/installing_vsphere/installing-vsphere.html#installation-vsphere-config-yaml_installing-vsphere

**Comment by flaper87** [comment]
/retest

lgtm but I will let @gamli75 the final word here

**Review by gamli75 — APPROVED** [review]

**Comment by YuviGold** [comment]
/hold
Please rebase

**Review by gamli75 — APPROVED** [review]

**Pull request merged**
